src/DeploymentProvider/AzureSpringCloudDeploymentProvider.ts

```diff
@@ -0,0 +1,207 @@
+        console.log('Deploy action');
+
+        let sourceType: string = this.determineSourceType(this.params.Package);
+
+        //If uploading a source folder, compress to tar.gz file.
+        let fileToUpload: string = sourceType == SourceType.SOURCE_DIRECTORY ?
+            await this.compressSourceDirectory(this.params.Package.getPath()) :
+            this.params.Package.getPath();
+
+
+        let deploymentName: string;
+        if (this.params.UseStagingDeployment) {
+            deploymentName = await dh.getStagingDeploymentName(this.client, this.params);
+
+            if (!deploymentName) { //If no inactive deployment exists
+                core.debug('No inactive deployment exists');
+                if (this.params.CreateNewDeployment) {
+                    core.debug('New deployment will be created');
+                    deploymentName = this.defaultInactiveDeploymentName; //Create a new deployment with the default name.
+                    this.params.DeploymentName = deploymentName;
+                } else
+                    throw Error('NoStagingDeploymentFound');
+            }
+        } else { //Deploy to deployment with specified name
+            core.debug('Deploying with specified name.');
+            deploymentName = this.params.DeploymentName;
+            let deploymentNames : Array<string> = await dh.getAllDeploymentsName(this.client, this.params);
+            if (!deploymentNames || !deploymentNames.includes(deploymentName)) {
+                core.debug(`Deployment ${deploymentName} does not exist`);
+                if (this.params.CreateNewDeployment) {
+                    if (deploymentNames.length > 1) {
+                        throw Error('TwoDeploymentsAlreadyExistCannotCreate' + deploymentName);
+                    } else {
+                        core.debug('Deployment will be created.');
+                    }
+                } else {
+                    throw Error('DeploymentDoesntExist' + deploymentName);
+                }
+
+            }
+        }
+        try {
+            console.log(`Action for service ${this.params.ServiceName} app ${this.params.AppName} deployment ${deploymentName}`);
+            await dh.deploy(this.client, this.params, sourceType, fileToUpload);
+            console.log('Deploy action successful');
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    /**
+     * Compresses sourceDirectoryPath into a tar.gz
+     * @param sourceDirectoryPath 
+     */
+    async compressSourceDirectory(sourceDirectoryPath: string): Promise<string> {
+        const fileName = `${uuidv4()}.tar.gz`;
+        core.debug(`CompressingSourceDirectory ${sourceDirectoryPath} ${fileName}`);
+        await tar.c({
+            gzip: true,
+            file: fileName,
+            sync: true,
+            cwd: sourceDirectoryPath,
+            onWarn: warning => {
+                core.warning(warning);
+            }
+        }, ['.']);
```

Comment:
We need to check whether we can skip certain paths from the archive, such as `.git`. You may refer to the CLI implementation.